python/CMakeLists.txt

```diff
@@ -265,11 +246,44 @@ message(STATUS "NumPy include dir: ${NUMPY_INCLUDE_DIRS}")
 
 include(UseCython)
 
-# PyArrow C++
+# Arrow C++ and set default PyArrow build options
 include(GNUInstallDirs)
-
 find_package(Arrow REQUIRED)
 
+# Top level cmake dir
+if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
```

Comment:
@kou for my education, could you explain what this `if` check is exactly doing or why we need it?

Reply:
It's for avoiding defining our options when this is used via `add_subdirectory()` (or [`FetchContent`](https://cmake.org/cmake/help/latest/module/FetchContent.html)).

We don't need this because:

* This is never used via `add_subdirectory()`
* Our option definitions are harmless with `add_subdirectory()` use